azkaban-hadoop-security-plugin/build.gradle

```diff
@@ -1,12 +1,20 @@
 apply plugin: 'distribution'
 
+configurations {
+    testCompile.extendsFrom compileOnly
```

Comment:
No, I don't think so. `testCompile` extends `compile` by default. But it doesn't seem to pull in from `compileOnly` which is why I added this line.

> IMHO, it is better to stay with the default/convention to avoid surprises as much as possible.

I find it intuitive to have the compileOnly classes available in my test classpath so that I can test my code! In fact, it is counter intuitive to me if my jar gets built successfully yet my tests don't compile.

Gradle configurations are built from file collections so that they can be extended, filtered and manipulated. So I don't think this should come as a surprise to a Gradle user. In fact, I think it might help to take a look at the task and configuration graph.
link: https://docs.gradle.org/3.3/userguide/java_plugin.html

> If someone forgets to add a dependency, tests will fail. Right? Why isn't it good enough?

This is good. But why do have to add it in 2 places is the question.